draft-irtf-cfrg-hpke.md

```diff
@@ -644,13 +679,60 @@ octet strings for public keys.
 
 # Security Considerations
 
-[[ TODO ]]
+The general security properties of HPKE are described in
+{{security-properties}}.  In this section, we consider a security issue that may
+arise in practice and an advanced use case.
+
+## Metadata Protection
+
+The authenticated modes of HPKE (PSK, Auth, AuthPSK) require that the receiver
+know what key material to use for the initiator.  This can be signaled in
+applications by sending the PSK ID (`pskID` above) and/or the initiator's public
+key (`pkI`).  However, these values themselves might be considered sensitive,
+since in a given application context, they might identify the initiator.
+
+An application that wishes to protect these metadata values without requiring
+further provisioning of keys can use an additional instance of HPKE, using the
+unauthenticated base mode.  Where the application might have sent `(pskID, pkI,
+enc, ciphertext)` before, it would now send (enc2, ciphertext2, enc, ciphertext),
+where `(enc2, ciphertext2)` represent the encryption of the `pskID` and `pkI`
+values.
+
+The cost of this approach is an additional KEM operation each for the sender and
+the receiver.  A potential lower-cost approach (involving only symmetric
+operations) would be available if the nonce-protection schemes in {{BNT19}}
+could be extended to cover other metadata.  However, this construction would
+require further analysis.
+
+## Designated-Verifier Signature
+
+The Auth and AuthPSK modes HPKE can be used to construct a lightweight
+"designated-verifier signature" scheme {{JKR96}}, in the sense that the message
+is authenticated as coming from the initiator, but the only party who can verify
+the authentication is the receiver (the holder of `skR`).
+
+To create such a signature, the initator simply performs a normal HPKE setup in
+the proper mode, and calls the Seal method on the resulting context with an
+empty plaintext value and the content to be signed as AAD.  This produces an
+encoded key `enc` and a ciphertext value that contains only the AAD tag.
+
+For example, using DHKEM-X25519 and AES-128-GCM, this would produce a 48-byte
+signature comprising a 32-byte ephemeral X25519 key and a 16-byte GCM tag.
+
+To verify such a signature, the receiver performs the corresponding HPKE setup
+and calls Open with the provided ciphertext.  If the AEAD authentication passes,
+then the signature is valid.
+
+This scheme is really just a re-use of the authentication scheme underlying the
```

Comment:
```suggestion
This scheme is re-uses the authentication scheme underlying the
```

Reply:
No, I think this is correct as-is -- e.g., with AES-GCM, we're not really using AES, we're using GMAC.

Reply:
I get that, but the suggestion doesn't change that fact? It's purely editorial.